src/components/NewsletterForm.jsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react'
+import styled, { keyframes } from 'styled-components'
+import useNewsletter from 'hooks/useNewsletter'
+import ArrowPathIcon from './icons/heroicons/ArrowPathIcon'
+import CheckIcon from './icons/heroicons/CheckIcon'
+
+const spin = keyframes`
+  from {
+    transform: rotate(0deg);
+  }
+  to {
+    transform: rotate(360deg);
+  }
+`
+
+const SpinningIcon = styled.div`
+  display: inline-flex;
+  animation: ${spin} 1s linear infinite;
+  svg {
+    width: 1.25rem;
+    height: 1.25rem;
+  }
+`
+
+const Input = styled.input`
```

Comment:
Out of curiosity, why didn't you use the Input component and override its style if needed?